### PR TITLE
Update remaining references to 5.5 on CI

### PR DESCRIPTION
**Pull request by j9liu**
## Description

I missed some spots in #1866 that still referred to Unreal 5.5; this PR addresses those.

## Issue number or link

N/A

## Author checklist

- [X] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [X] I have done a full self-review of my code.
- ~[ ] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).~
- ~[ ] I have added or updated unit tests to ensure consistent code coverage as necessary.~
- ~[ ] I have updated the documentation as necessary.~

## Testing plan

N/A since this mostly affects CI.

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

**Comment by j9liu** [comment]
**EDIT**: resolved, had to delete my `.ezvcpkg` directory.

-----------------

For some reason when I try a clean build locally, I'm getting an error where `tinyxml` keeps trying to use `5.5`:

```
CMake Error in cesium-native/CesiumRasterOverlays/CMakeLists.txt:
  Imported target "tinyxml2::tinyxml2" includes non-existent path

    "C:/Program Files/Epic Games/UE_5.5/Engine/Source/ThirdParty/TinyXML2/9.0.0/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

I can't tell where this variable is being resolved though, because I definitely changed the `UNREAL_ENGINE_ROOT` variable, and this happens even in a clean build...

**Comment by j9liu** [comment]
**EDIT:** I think this now has to do with the fact that I have 14.44 and 14.38 installed and ezvcpkg is using the latest version.

-------------------------

Still running into errors... this time at the linker stage:

```
27>[29/31] Link [x64] UnrealEditor-CesiumEditor-Win64-DebugGame.dll
27>ktx.lib(basisu_comp.cpp.obj) : error LNK2019: unresolved external symbol __std_find_last_trivial_1 referenced in function "char const * __cdecl std::_Find_last_vectorized<char const ,char>(char const * const,char const * const,char)" (??$_Find_last_vectorized@$$CBDD@std@@YAPEBDQEBD0D@Z)
27>ktx.lib(basisu_gpu_texture.cpp.obj) : error LNK2001: unresolved external symbol __std_find_last_trivial_1
27>ada.lib(ada.cpp.obj) : error LNK2001: unresolved external symbol __std_find_last_trivial_1
27>spdlogd.lib(spdlog.cpp.obj) : error LNK2019: unresolved external symbol __std_find_first_of_trivial_pos_1 referenced in function "unsigned __int64 __cdecl std::_Find_first_of_pos_vectorized<char,char>(char const * const,unsigned __int64,char const * const,unsigned __int64)" (??$_Find_first_of_pos_vectorized@DD@std@@YA_KQEBD_K01@Z)
27>absl_strings.lib(match.cc.obj) : error LNK2001: unresolved external symbol __std_find_first_of_trivial_pos_1
27>ada.lib(ada.cpp.obj) : error LNK2001: unresolved external symbol __std_find_first_of_trivial_pos_1
27>spdlogd.lib(spdlog.cpp.obj) : error LNK2019: unresolved external symbol __std_find_last_of_trivial_pos_1 referenced in function "unsigned __int64 __cdecl std::_Find_last_of_pos_vectorized<char,char>(char const * const,unsigned __int64,char const * const,unsigned __int64)" (??$_Find_last_of_pos_vectorized@DD@std@@YA_KQEBD_K01@Z)
27>absl_log_internal_message.lib(log_message.cc.obj) : error LNK2001: unresolved external symbol __std_find_last_of_trivial_pos_1
27>ada.lib(ada.cpp.obj) : error LNK2019: unresolved external symbol __std_search_1 referenced in function "char const * __cdecl std::_Search_vectorized<char const ,char const >(char const * const,char const * const,char const * const,unsigned __int64)" (??$_Search_vectorized@$$CBD$$CBD@std@@YAPEBDQEBD00_K@Z)
27>C:\Dev\cesium-unreal-samples\Plugins\cesium-unreal\Binaries\Win64\UnrealEditor-CesiumEditor-Win64-DebugGame.dll : fatal error LNK1120: 4 unresolved externals
27>[30/31] Link [x64] UnrealEditor-CesiumRuntime-Win64-DebugGame.dll
27>ktx.lib(basisu_comp.cpp.obj) : error LNK2019: unresolved external symbol __std_find_last_trivial_1 referenced in function "char const * __cdecl std::_Find_last_vectorized<char const ,char>(char const * const,char const * const,char)" (??$_Find_last_vectorized@$$CBDD@std@@YAPEBDQEBD0D@Z)
27>ktx.lib(basisu_gpu_texture.cpp.obj) : error LNK2001: unresolved external symbol __std_find_last_trivial_1
27>ada.lib(ada.cpp.obj) : error LNK2001: unresolved external symbol __std_find_last_trivial_1
27>spdlogd.lib(spdlog.cpp.obj) : error LNK2019: unresolved external symbol __std_find_first_of_trivial_pos_1 referenced in function "unsigned __int64 __cdecl std::_Find_first_of_pos_vectorized<char,char>(char const * const,unsigned __int64,char const * const,unsigned __int64)" (??$_Find_first_of_pos_vectorized@DD@std@@YA_KQEBD_K01@Z)
27>absl_strings.lib(match.cc.obj) : error LNK2001: unresolved external symbol __std_find_first_of_trivial_pos_1
27>ada.lib(ada.cpp.obj) : error LNK2001: unresolved external symbol __std_find_first_of_trivial_pos_1
27>spdlogd.lib(spdlog.cpp.obj) : error LNK2019: unresolved external symbol __std_find_last_of_trivial_pos_1 referenced in function "unsigned __int64 __cdecl std::_Find_last_of_pos_vectorized<char,char>(char const * const,unsigned __int64,char const * const,unsigned __int64)" (??$_Find_last_of_pos_vectorized@DD@std@@YA_KQEBD_K01@Z)
27>absl_log_internal_message.lib(log_message.cc.obj) : error LNK2001: unresolved external symbol __std_find_last_of_trivial_pos_1
27>spz.lib(load-spz.cc.obj) : error LNK2019: unresolved external symbol __std_search_1 referenced in function "char const * __cdecl std::_Search_vectorized<char const ,char const >(char const * const,char const * const,char const * const,unsigned __int64)" (??$_Search_vectorized@$$CBD$$CBD@std@@YAPEBDQEBD00_K@Z)
27>ada.lib(ada.cpp.obj) : error LNK2001: unresolved external symbol __std_search_1
27>spz.lib(load-spz.cc.obj) : error LNK2019: unresolved external symbol __std_find_end_1 referenced in function "char const * __cdecl std::_Find_end_vectorized<char const ,char const >(char const * const,char const * const,char const * const,unsigned __int64)" (??$_Find_end_vectorized@$$CBD$$CBD@std@@YAPEBDQEBD00_K@Z)
27>C:\Dev\cesium-unreal-samples\Plugins\cesium-unreal\Binaries\Win64\UnrealEditor-CesiumRuntime-Win64-DebugGame.dll : fatal error LNK1120: 5 unresolved externals
```
Yes, I'm positive I'm generating the project build files with 5.6 as the selected version 😬

File: .github/workflows/buildApple.yml
Comment (azrogers) [inline .github/workflows/buildApple.yml]: If this is just a pre-Unreal 5.5 thing, can we just remove the bit where we put it in `~/.config` in the first place and write it directly to `~/Documents` instead?

File: .github/workflows/buildApple.yml [hunk 1]
```diff
@@ -138,7 +138,7 @@ jobs:
           rm -rf extern
       - name: Build plugin for macOS and iOS
         run: |
-          sed -i '' 's/\"EngineVersion\": \"5.5.0\"/\"EngineVersion\": \"${{ inputs.unreal-engine-version }}\"/g' CesiumForUnreal.uplugin
+          sed -i '' 's/\"EngineVersion\": \"5.6.0\"/\"EngineVersion\": \"${{ inputs.unreal-engine-version }}\"/g' CesiumForUnreal.uplugin
```
Comment (azrogers) [inline .github/workflows/buildApple.yml]: You don't necessarily have to change this, it might be a hassle to get working, but you might be able to change this to `sed -i '' 's/\"EngineVersion\": \"5.[[:digit:]]\+.0\"/\"EngineVersion\": \"${{ inputs.unreal-engine-version }}\"/g' CesiumForUnreal.uplugin` so that we don't have to keep manually changing the number every time we change the uplugin file.

File: .github/workflows/buildWindows.yml [hunk 2]
```diff
@@ -112,7 +112,7 @@ jobs:
           rm -r -fo extern
       - name: Overwrite plugin engine version
         run: |
-          ((Get-Content -path CesiumForUnreal.uplugin -Raw) -replace '"EngineVersion": "5.5.0"','"EngineVersion": "${{ inputs.unreal-engine-version }}"') | Set-Content -Path CesiumForUnreal.uplugin
+          ((Get-Content -path CesiumForUnreal.uplugin -Raw) -replace '"EngineVersion": "5.6.0"','"EngineVersion": "${{ inputs.unreal-engine-version }}"') | Set-Content -Path CesiumForUnreal.uplugin
```
Comment (azrogers) [inline .github/workflows/buildWindows.yml]: I believe the powershell equivalent of my above `sed` suggestion would be `((Get-Content -path CesiumForUnreal.uplugin -Raw) -replace '"EngineVersion": "5.\d+.0"','"EngineVersion": "${{ inputs.unreal-engine-version }}"') | Set-Content -Path CesiumForUnreal.uplugin` or something like that.

**Comment by azrogers** [comment]
@j9liu Suggested a few possible improvements, but broadly it looks good. Also, I believe the title of this PR should be "Update remaining references to **5.5** on CI"

File: .github/workflows/buildApple.yml
Comment (j9liu) [reply .github/workflows/buildApple.yml]: Good callout!

**Comment by j9liu** [comment]
Thanks @azrogers for the review! I worked in your suggestions and they seem to work well. Ready for another look!

**Comment by azrogers** [comment]
Glad they work! Could you apply the same changes to `buildLinux.yml` and `buildAndroid.yml`? If you can do that and CI passes then I think it's good to merge! @j9liu

**Comment by j9liu** [comment]
@azrogers done!

**Comment by azrogers** [comment]
Thanks @j9liu!
